symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/stream/StreamService.java

```diff
@@ -0,0 +1,288 @@
+package com.symphony.bdk.core.service.stream;
+
+import com.symphony.bdk.core.api.invoker.ApiException;
+import com.symphony.bdk.core.api.invoker.ApiRuntimeException;
+import com.symphony.bdk.core.auth.AuthSession;
+import com.symphony.bdk.core.service.stream.constant.AttachmentSort;
+import com.symphony.bdk.gen.api.StreamsApi;
+import com.symphony.bdk.gen.api.model.RoomDetail;
+import com.symphony.bdk.gen.api.model.Stream;
+import com.symphony.bdk.gen.api.model.StreamAttachmentItem;
+import com.symphony.bdk.gen.api.model.StreamAttributes;
+import com.symphony.bdk.gen.api.model.StreamFilter;
+import com.symphony.bdk.gen.api.model.V2AdminStreamFilter;
+import com.symphony.bdk.gen.api.model.V2AdminStreamList;
+import com.symphony.bdk.gen.api.model.V2MembershipList;
+import com.symphony.bdk.gen.api.model.V2RoomSearchCriteria;
+import com.symphony.bdk.gen.api.model.V2StreamAttributes;
+import com.symphony.bdk.gen.api.model.V3RoomAttributes;
+import com.symphony.bdk.gen.api.model.V3RoomDetail;
+import com.symphony.bdk.gen.api.model.V3RoomSearchResults;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Service class for managing streams.
+ *
+ * This service is used for retrieving information about a particular stream or
+ * chatroom, searching streams, listing members, attachments of a particular stream,
+ * perform some action related to a stream like:
+ * <p><ul>
+ *   <li>Create a IM or MIM</li>
+ *   <li>Create a chatroom</li>
+ *   <li>Activate or Deactivate a chatroom</li>
+ *   <li></li>
+ * </ul></p>
+ *
+ */
+@Slf4j
+public class StreamService {
```

Comment:
Missing endpoints put in the STREAMS part of https://developers.symphony.com/restapi/reference:

- https://developers.symphony.com/restapi/reference#room-members
- https://developers.symphony.com/restapi/reference#add-member
- https://developers.symphony.com/restapi/reference#remove-member
- https://developers.symphony.com/restapi/reference#share-v3
- https://developers.symphony.com/restapi/reference#promote-owner
- https://developers.symphony.com/restapi/reference#demote-owner

Reply:
Opened https://perzoinc.atlassian.net/browse/APP-3033 to track further development of the StreamService.